samples/contrib/aws-samples/README.md

```diff
@@ -0,0 +1,198 @@
+
+###################################################################
+# This is the only thing that you need to change to run this code 
+# Give the name of your S3 bucket 
+bucket = '<bucket-name>' 
+
+# If you are gonna use the default values of the pipeline then 
+# give a bucket name which is in us-east-1 region 
+###################################################################
+
+
+# Load the dataset
+urllib.request.urlretrieve("http://deeplearning.net/data/mnist/mnist.pkl.gz", "mnist.pkl.gz")
+with gzip.open('mnist.pkl.gz', 'rb') as f:
+    train_set, valid_set, test_set = pickle.load(f, encoding='latin1')
+
+
+# Upload dataset to S3
+from sagemaker.amazon.common import write_numpy_to_dense_tensor
+import io
+import boto3
+
+train_data_key = 'mnist_kmeans_example/train_data'
+test_data_key = 'mnist_kmeans_example/test_data'
+train_data_location = 's3://{}/{}'.format(bucket, train_data_key)
+test_data_location = 's3://{}/{}'.format(bucket, test_data_key)
+print('training data will be uploaded to: {}'.format(train_data_location))
+print('training data will be uploaded to: {}'.format(test_data_location))
+
+# Convert the training data into the format required by the SageMaker KMeans algorithm
+buf = io.BytesIO()
+write_numpy_to_dense_tensor(buf, train_set[0], train_set[1])
+buf.seek(0)
+
+boto3.resource('s3').Bucket(bucket).Object(train_data_key).upload_fileobj(buf)
+
+# Convert the test data into the format required by the SageMaker KMeans algorithm
+write_numpy_to_dense_tensor(buf, test_set[0], test_set[1])
+buf.seek(0)
+
+boto3.resource('s3').Bucket(bucket).Object(test_data_key).upload_fileobj(buf)
+
+# Convert the valid data into the format required by the SageMaker KMeans algorithm
+numpy.savetxt('valid-data.csv', valid_set[0], delimiter=',', fmt='%g')
+s3_client = boto3.client('s3')
+input_key = "{}/valid_data.csv".format("mnist_kmeans_example/input")
+s3_client.upload_file('valid-data.csv', bucket, input_key)
+```
+Run this file `python s3_sample_data_creator.py`
+
+### Role Input
+
+This role is used by SageMaker jobs created by the KFP to access the S3 buckets and other AWS resources.
+Run these commands to create the sagemaker-execution-role.   
+Note down the Role ARN. You need to give this Role ARN as input in pipeline.
+
+```
+TRUST="{ \"Version\": \"2012-10-17\", \"Statement\": [ { \"Effect\": \"Allow\", \"Principal\": { \"Service\": \"sagemaker.amazonaws.com\" }, \"Action\": \"sts:AssumeRole\" } ] }"
+aws iam create-role --role-name kfp-example-sagemaker-execution-role --assume-role-policy-document "$TRUST"
+aws iam attach-role-policy --role-name kfp-example-sagemaker-execution-role --policy-arn arn:aws:iam::aws:policy/AmazonSageMakerFullAccess
+aws iam attach-role-policy --role-name kfp-example-sagemaker-execution-role --policy-arn arn:aws:iam::aws:policy/AmazonS3FullAccess
+aws iam get-role --role-name kfp-example-sagemaker-execution-role --output text --query 'Role.Arn'
+
+# note down the Role ARN. 
+```
+
```

Comment:
One additional feature I might add to the README (to preempt some inevitable questions) would be an FAQ. I imagine some people will be confused about why their authentication is failing (some common error messages) and we would be able to provide links to instructions about configuration options.

Reply:
let me create a new PR for that